routes/api/users.js

```diff
@@ -171,4 +173,115 @@ router.post('/users/verify/resend-email', (req, res, next) => {
   return null;
 });
 
+
+router.post('/users/forgot-password', (req, res, next) => {
+  const { email } = req.body.user;
+  if (email === undefined) {
+    return res.status(400)
+      .json({ message: 'Email can not be undefined' });
+  }
+  if (email === '') {
+    return res.status(400)
+      .json({ message: 'Email can not be empty' });
+  }
+  User.findOne({
+    where: { email },
+  })
+    .then((user) => {
+      if (!user) {
+        return res.status(404)
+          .json({ message: 'Email not found!' });
+      }
+
+      const token = tokenService
+        .generateToken({ id: user.id, username: user.username }, 60 * 30);
+      let url = `${process.env.BASEURL}`;
+      url += `/api/users/reset-password?token=${token}`;
+      const mailOptions = emailService.mailOptions(
+        user.email,
+        "Authors Heaven's account token",
+        emailTemplates.resetPasswordTemplate(url, user),
+      );
+      emailService.sendMail(mailOptions)
+        .then(() => res.status(200).json({
+          message: 'Check your email for password reset token',
+        }))
+        .catch(next);
+      return null;
+    })
+    .catch(next);
+  return null;
+});
+
+
+router.get('/users/reset-password', (req, res) => {
+  const { token } = req.query;
+  const decoded = tokenService.verifyToken(token);
+
+  if (!decoded) {
+    return res.status(401)
+      .json({ message: 'Invalid token' });
+  }
+
+  return res.status(200)
+    .json({ message: 'Token verified, You can now reset your password' });
```

Comment:
I think the message should be something like `Verification Successful. You can now reset your password`

Reply:
Ok, Implemented.